packages/devtools_app/lib/src/screens/memory/primitives/class_name.dart

```diff
@@ -0,0 +1,285 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../../../primitives/simple_items.dart';
+
+@immutable
+class HeapClassName {
+  HeapClassName({required this.className, required library})
+      : library = _normalizeLibrary(library) {
+    assert(
+      !isCore || !isDartOrFlutter,
+      'isCore and isDartOrFlutter must be exclusive',
+    );
+  }
+
+  HeapClassName.fromClassRef(ClassRef? classRef)
+      : this(
+          library: _library(
+            classRef?.library?.name,
+            classRef?.library?.uri,
+          ),
+          className: classRef?.name ?? '',
+        );
+
+  HeapClassName.fromHeapSnapshotClass(HeapSnapshotClass? theClass)
+      : this(
+          library: _library(
+            theClass?.libraryName,
+            theClass?.libraryUri.toString(),
+          ),
+          className: theClass?.name ?? '',
+        );
+
+  static String _library(String? libName, String? libUrl) {
+    libName ??= '';
+    if (libName.isNotEmpty) return libName;
+    return libUrl ?? '';
+  }
+
+  final String className;
+  final String library;
+
+  String get fullName => library.isNotEmpty ? '$library/$className' : className;
+
+  bool get isSentinel => className == 'Sentinel' && library.isEmpty;
+
+  /// Detects if a class can retain an object from garbage collection.
```

Comment:
nit: can we phrase this comment in terms of the getter name? This dart doc would make me think the opposite, that a weak entry retains an object from garbage collection. How about:

/// Whether a class can hold a reference to an object without preventing garbage collection.